dali/operators/python_function/dltensor_function.h

```diff
@@ -193,6 +193,7 @@ class DLTensorPythonFunctionImpl : public Operator<Backend> {
     std::lock_guard<std::mutex> operator_guard(operator_lock);
     py::gil_scoped_acquire interpreter_guard{};
     py::object output_o = py::none();
+    auto curr_batch_size = ws.GetInputBatchSize(0);
```

Comment:
I don't think we can safely assume that there is Input 0 in this Operator.
It certainly requires more logic and probably in some cases would behave as supporting requested batch size which would result in some refactoring. Maybe for such problematic operators we can add an explicit check that it doesn't work with variable batch size and fail for cases where the BS != max_batch_size. That way we could quickly identify them and it would be probably fine to have several of such operators slowly getting updated.

Reply:
I've put better batch size integrity checking, don't hesitate to take a look ;)